openff/toolkit/typing/engines/smirnoff/parameters.py

```diff
@@ -342,10 +342,14 @@ class ParameterAttribute:
     def __init__(
         self,
         default: Any = UNDEFINED,
-        unit: Optional[Unit] = None,
+        unit: Unit | str | None = None,
         converter: Optional[Callable] = None,
         docstring: str = "",
     ):
+        if isinstance(unit, str):
+            # be careful with module & variable names
+            unit = Unit(unit)
```

Comment:
Another option is simply
```suggestion
        if isinstance(unit, str):
            raise ValueError(f"Input argument {unit} must be `Unit` or `None`")
```

Either fixes my use case

Reply:
(not blocking) I'm in favor of the current fix, but out of curiosity, did you ever figure out what was going on with the trying-to-treat-a-string-as-a-quantity error arising even when a quantity was explicitly given?